skimage/filters/thresholding.py

```diff
@@ -818,27 +819,40 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            :DOI:`10.1117/12.767755`
     """
-    if w == 1 or w % 2 == 0:
-        raise ValueError(
-            "Window size w = %s must be odd and greater than 1." % w)
+    def _sanity_check_window_size(axis_size):
+        if axis_size == 1 or axis_size % 2 == 0:
+            raise ValueError(
+                "Window size w = {} must be odd and greater than 1."
+                .format(axis_size)
+            )
+
+    if isinstance(w, tuple):
```

Comment:
here it might be better to check for `isinstance(w, collections.Iterable)`, so that people can pass in a list. In many APIs they are used interchangeably.